### PR TITLE
feat: add `sync` variant

### LINKED ISSUE #10876: A command field in astro:config:setup returns build value while running check command
### Astro Info

```block
Astro                    v4.7.0
Node                     v18.18.0
System                   Linux (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             test
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

Let's assume an Astro config like below:
```js
export default defineConfig({
  integrations: [
    {
      name: 'test',
      hooks: {
        'astro:config:setup': ({ command }) => {
          console.log(command);
        },
      },
    },
  ],
});
```
While running `astro check` the console outputs `build` string. This can be confusing, especially when running the command in the default pipeline: `astro check & astro build`. In this case, the `astro:config:setup` hook is called twice with `command === 'build'`, unexpectedly triggering certain code twice during the build.
Also, this behavior isn't documented: [https://docs.astro.build/en/reference/integrations-reference/#command-option](https://docs.astro.build/en/reference/integrations-reference/#command-option)

It appears to be caused by a hardcoded 'build' value in the mentioned line:[https://github.com/withastro/astro/blob/main/packages/astro/src/core/sync/index.ts#L62](https://github.com/withastro/astro/blob/main/packages/astro/src/core/sync/index.ts#L62)
```ts
const settings = await runHookConfigSetup({
	settings: _settings,
	logger: logger,
	command: 'build', // <- here
});
```

### What's the expected result?

I suggest three possible solutions:

1. `command` while running `astro check` has `check` value
2. `astro check` command is integrated with `astro build` always, so `astro:config:setup` calls once only (I think it's the worst option)
3. Describe this behavior in documentation

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-yfhwgr?file=astro.config.mjs

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes #10876 

I added a `"sync"` variant to the command. This will be useful to the extension too.

This is flagged as minor, because it adds new information. 

## Testing

Existing CI should pass. Should I create a test? 🤔 

## Docs

https://github.com/withastro/docs/pull/9110

### CODE REVIEW

**Review by florian-lefebvre — APPROVED** [review]
LGTM, may break a few integrations

**Comment by ematipico** [comment]
> LGTM, may break a few integrations

How? :| 

Maybe because they use a `switch` of `command`?

**Comment by florian-lefebvre** [comment]
yes, command have been holding 3 possible values for so long that maybe some conditions will be broken. That being said, I don't see a better way than this so up to integrations authors to update

**Review by bluwy — APPROVED** [review]

File: .changeset/afraid-apricots-develop.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+The hook `astro:config:setup` now provide a `"sync"` variant for the `command` property. This value is usually set when calling the command `astro sync`
```
Comment (sarah11918) [inline .changeset/afraid-apricots-develop.md:5]:
```suggestion
Adds a new variant `sync` for the `astro:config:setup` hook's `command` property. Set this value to call the command `astro sync`.
```
I wasn't sure whether this was something that the integration author configures, or whether it "gets" set by Astro on them, but if the integration author sets this, I might phrase it more about what they need to do.

And "usually" in docs makes me nervous! :sweat_smile: Is there a particular concern/case where this wouldn't happen/work? Or do you just mean that this is "common usage" (in which case, I don't think it's necessary to say usually)

**Review by sarah11918 — COMMENTED** [review]
Quick suggestion for the changeset @ematipico ! When something new is added, I always like starting with "Adds" because that's the biggest clue that this is new, something you *could* check out, but not necessary and shouldn't break anything.

Also, I saw the conversation here about whether or not some projects might be relying on the fact that there have been exactly three options previously. If you felt it was necessary or helpful, you *could* consider adding another paragraph with a single line just saying something like:

```
If your integration previously relied on knowing how many variants existed for the `command` property, you must update your logic to account for this new option.
```

if you feel like that's a common enough thing!

File: .changeset/afraid-apricots-develop.md (hunk 1 above)
Comment (florian-lefebvre) [reply .changeset/afraid-apricots-develop.md:5]: Astro sets it so that integration authors know their integration run while a user executes `astro sync`

Comment (sarah11918) [reply .changeset/afraid-apricots-develop.md:5]: Thanks for the clarification, Florian! Then the existing wording is great, but *usually* still makes me nervous? Would you say that's a key word to include here?

Comment (florian-lefebvre) [reply .changeset/afraid-apricots-develop.md:5]: No it's confusing, lets remove it

**Comment by florian-lefebvre** [comment]
I think it's useful to add such paragraph

File: .changeset/afraid-apricots-develop.md (hunk 1 above)
Comment (sarah11918) [reply .changeset/afraid-apricots-develop.md:5]: OK, I updated my suggestion above taking all this into account!

**Review by sarah11918 — APPROVED** [review]
Approving for docs!

**Pull request merged**